hail/src/main/scala/is/hail/methods/LocalLDPrune.scala

```diff
@@ -274,23 +276,47 @@ object LocalLDPrune {
     })
   }
 
-  def apply(mt: MatrixTable, callField: String = "GT", r2Threshold: Double = 0.2, windowSize: Int = 1000000, maxQueueSize: Int): Table = {
+  def apply(mt: MatrixTable,
+    callField: String = "GT", r2Threshold: Double = 0.2, windowSize: Int = 1000000, maxQueueSize: Int
+  ): Table = {
+    val pruner = LocalLDPrune(callField, r2Threshold, windowSize, maxQueueSize)
+    val mv = mt.value
+    new Table(HailContext.get, TableLiteral(pruner.execute(mt.value)))
```

Comment:
This seems like it should be (mv instead of .  mt.value)

```scala
new Table(HailContext.get, TableLiteral(pruner.execute(mv))
```

Reply:
Also, how sure are we of Scala's inliner. Afaik at least some implementations depend on function complexity. What do we gain by the indirection?

Regarding inline efficiency, besides performance tests, would it be interesting to inspect bytecode diffs in cases of regression? For C it would be interesting to spend a bit of time looking at assembly outputs. Maybe that's a dumb use of time, although it seems like knowing why performance changes is useful, and could help us increase depth of reason in choosing abstractions. In this case it's likely not a performance concern, but I don't feel confident . in that guess.

Reply:
No idea on Scala's inliner.  We spent a lot of time trying to understand and make Scala performant.  Basically,  we've given up.  We've read a lot of bytecode, although it's relationship to performance is complicated because the JIT can improve things a lot.  I'm not inclined to invest in tooling like this.  Scala's specialized is a total mess and known to generate bad code (quirks, ha): http://aleksandar-prokopec.com/2013/11/03/specialization-quirks.html

For C/C++, yeah, looking at generated code can definitely be useful.

Reply:
Thank you